Libraries/LLM/Load.swift

```diff
@@ -11,82 +11,91 @@ struct LLMError: Error {
     let message: String
 }
 
-/// Load and return the model and tokenizer
-public func load(
-    hub: HubApi = HubApi(), configuration: ModelConfiguration,
-    progressHandler: @Sendable @escaping (Progress) -> Void = { _ in }
-) async throws -> (LLMModel, Tokenizer) {
+func prepareModelDirectory(
+    hub: HubApi, configuration: ModelConfiguration,
+    progressHandler: @Sendable @escaping (Progress) -> Void
+) async throws -> URL {
```

Comment:
This is the only part of model loading that is asynchronous and it just produces a URL (with the downloaded weights).